phoenix-core/src/main/java/org/apache/phoenix/schema/types/FileSystemBasedLobStore.java

```diff
@@ -21,41 +21,62 @@
 import com.google.common.annotations.VisibleForTesting;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.RandomStringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.FileSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.io.FileInputStream;
-import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
 import java.io.InputStream;
 
 /**
  * Test Implementation to unblock others writes to /tmp/
  */
 public class FileSystemBasedLobStore implements LobStore {
 
-    private static final String ROOT_DIRECTORY = "/tmp/";
+    public static final Logger LOGGER =
+            LoggerFactory.getLogger(FileSystemBasedLobStore.class);
+    private static final Path ROOT_DIRECTORY = new Path("/user/hdfs/lob/hdfs/");
 
+    private FileSystem initializeHDFS() throws IOException {
+        Configuration conf = new Configuration();
+        String hdfsuri = "hdfs://localhost:9000";
+        conf.set("fs.defaultFS", hdfsuri);
+        //==== Create folder if not exists
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+        if(!fs.exists(ROOT_DIRECTORY)) {
+            fs.mkdirs(ROOT_DIRECTORY);
+            LOGGER.info("Path "+ ROOT_DIRECTORY +" created.");
+        }
+        return fs;
+    }
 
     @Override
     public String putLob(InputStream lobStream) throws LobStoreException {
         String lobLocator = RandomStringUtils.randomAlphanumeric(15); //For now use lobLocator
-        try (FileOutputStream out = new FileOutputStream(getPathFromLocator(lobLocator)) ){
-            //Write the inputstream to our file
-            IOUtils.copy(lobStream,out);
-            //TODO: close lobStream?
+        Path hdfswritepath = new Path(ROOT_DIRECTORY + "/" + lobLocator);
+        try (FileSystem fs = initializeHDFS();
+             OutputStream os = fs.create(hdfswritepath)){
```

Comment:
We should probably do a quick "if not exists" kind of check to make sure we aren't overwriting an existing file